server/etcdserver/api/etcdhttp/base.go

```diff
@@ -40,24 +43,46 @@ func HandleBasic(lg *zap.Logger, mux *http.ServeMux, server etcdserver.ServerPee
 	mux.HandleFunc(versionPath, versionHandler(server.Cluster(), serveVersion))
 }
 
-func versionHandler(c api.Cluster, fn func(http.ResponseWriter, *http.Request, string)) http.HandlerFunc {
+func HandleBasicForV3(lg *zap.Logger, mux *http.ServeMux, server serverversion.Server) {
+	mux.HandleFunc(varsPath, serveVars)
+	mux.HandleFunc(versionPath, versionHandlerV3(server, serveVersion))
```

Comment:
Why we need to pass `serveVersion` here as a function parameter.  I don't see this method wrapping other functions. I think that functional programming that this code is currently performing is badly hitting readability. 

If we need multiple implementations, we should use interface instead. But I don't see purpose for the multiple implementations.

Reply:
Yes, I agree. Actually @serathius raised a [similar comment](https://github.com/etcd-io/etcd/pull/13772#discussion_r823483072) above. 

Please note that there is a separate PR [13655](https://github.com/etcd-io/etcd/pull/13655) for the cleanup of the v2 implementation. Once the PR and this one are merged, I will refactor the code, either combining the V2 and V3 implementations or using an interface.

Reply:
Resolved.